File: packages/core/tests/fixtures/unhealthy/hallucinated-imports.ts
/**
 * unhealthy/hallucinated-imports.ts
 *
 * Fixture for HallucinatedPackageImport detection.
 * Contains one real npm package (openai) and one obviously-fake package (xyzzy-not-a-real-package-abc123).
 * Used in packages/core/tests/analyzers/hallucinated-import.test.ts.
 */

// Real package — should NOT trigger HallucinatedPackageImport
import OpenAI from 'openai';

// Fake package — should trigger HallucinatedPackageImport
import { something } from 'xyzzy-not-a-real-package-abc123';

export function callFakePackage() {
  const client = new OpenAI({ apiKey: process.env.OPENAI_API_KEY });
  // @ts-ignore — something is from the fake package
  return something(client);
}

File: packages/core/tests/fixtures/unhealthy/hallucinated-imports.py
"""
unhealthy/hallucinated-imports.py

Fixture for HallucinatedPackageImport detection in Python.
Contains one real PyPI package (requests) and one obviously-fake package (xyzzy_not_a_real_package_abc123).
Used in packages/core/tests/analyzers/hallucinated-import.test.ts.
"""

# Real package — should NOT trigger HallucinatedPackageImport
import requests

# Fake package — should trigger HallucinatedPackageImport
import xyzzy_not_a_real_package_abc123


def fetch_data(url: str) -> dict:
    response = requests.get(url)
    result = xyzzy_not_a_real_package_abc123.process(response.json())
    return result
